Public/Src/Utilities/Native/IO/Helpers.cs

```diff
@@ -135,12 +138,70 @@ public static class Helpers
                     // Only log exceptions if they are not being re-thrown already
                     if (logExceptions)
                     {
-                        Tracing.Logger.Log.RetryOnFailureException(Events.StaticContext, e.Message ?? e.ToString());
+                        Tracing.Logger.Log.RetryOnFailureException(Events.StaticContext, e.ToString());
                     }
                 }
             }
 
             return done;
         }
+
+        /// <summary>
+        /// A generic version of <see cref="RetryOnFailure"/>.
+        /// </summary>
+        public static Possible<TResult, Failure> RetryOnFailure<TResult>(

```

Comment:
>RetryOnFailure [](start = 49, length = 14)

Should the RetryOnFilaure from above be implemented using this one?

Reply:
never mind, this is not async....

---
In reply to: [273279329](https://github.com/Microsoft/BuildXL/pull/104#discussion_r273279329) [](ancestors = 273279329)